### ISSUE TITLE
Failure on OLCF system due to warnings sent to stderr by compiler wrappers?

### ISSUE THREAD

**Issue by BarrySmith** [spack/spack#3144]
```
- Executing: gcc -E  -I/tmp/petsc-GcoxZn/config.setCompilers  /tmp/petsc-GcoxZn/config.setCompilers/conftest.c 
Possible ERROR while running preprocessor: exit code 256
stderr:
/autofs/nccs-svm1_home1/mbt/spack/lib/spack/env/gcc/gcc: line 358: /autofs/nccs-svm1_home1/mbt/spack/bin/spack-cc-petsc@develop%gcc@5.3.0~boost~complex~debug+double+hdf5+hypre~int64+metis+mpi~mumps+shared+superlu-dist+trilinos arch=cray-CNL-interlagos /n4yqikl.in.log: No such file or directory
/autofs/nccs-svm1_home1/mbt/spack/lib/spack/env/gcc/gcc: line 359: /autofs/nccs-svm1_home1/mbt/spack/bin/spack-cc-petsc@develop%gcc@5.3.0~boost~complex~debug+double+hdf5+hypre~int64+metis+mpi~mumps+shared+superlu-dist+trilinos arch=cray-CNL-interlagos /n4yqikl.out.log: No such file or directory
```
Any idea why spack is producing this warning message to stderr?

**Comment by BarrySmith** [comment]
Mark Berrill has found the bug and a fix. 

Here is the change that seems to fix the issue.  In lib/spack/spack/spec.py on line 2459 I changed:
```
                   out.write('/' + fmt % (self.dag_hash(7)))
to
                   out.write('-' + fmt % (self.dag_hash(7)))
```
In the comments for that function I believe this should be the correct behavior.  I think the fact that the hash started with an 'n' causing the environmental variable "SPACK_SHORT_SPEC" to have "/n" in it may have been causing the compiler to treat the hash as a command on a new line.  After the change, PETSc configured successfully and is currently building.

**Comment by adamjstewart** [comment]
Newline character would be `\n`, not `/n`. It must be interpreting the hash as a directory. The fix looks right to me, but @scheibelp may know better. This line was introduced in #2941.

**Comment by BarrySmith** [comment]
Yes, you are right about the newline business.

**Comment by scheibelp** [comment]
Sorry for the redirect but @becker33 may have some insight on the impact of replacing '/' with '-'

**Comment by adamjstewart** [comment]
@scheibelp Ah yes, sorry. Your GitHub account pictures look so similar that I always mix you two up 😆

**Comment by alalazo** [comment]
What's the status of this? Has this been fixed in #3928?

**Comment by alalazo** [comment]
Closing for lack of reply. Feel free to reopen if needed.
